orc8r/gateway/python/magma/magmad/config_manager.py

```diff
@@ -145,4 +146,29 @@ def did_mconfig_change(serv_name):
             if srv in mconfig.configs_by_key:
                 configs_by_key[srv] = mconfig.configs_by_key.get(srv)
 
+        # version should be in 1.1.1 format with only positive numbers allowed
+        VERSION_REGEX = re.compile(r"[0-9]+\.(?P<minor_version>[0-9]+)\.[0-9]+")
+
+        agw_version = self._magmad_service.version
+        unpacked_mconfig = mconfig.configs_by_key.get(MAGMAD)
+        if unpacked_mconfig.Is(mconfigs_pb2.MagmaD.DESCRIPTOR):
+            magmad_parsed = mconfigs_pb2.MagmaD()
+            unpacked_mconfig.Unpack(magmad_parsed)
+            orc8r_version = magmad_parsed.orc8r_version
+
+            agw_version_parsed = VERSION_REGEX.match(agw_version)
+            orc8r_version_parsed = VERSION_REGEX.match(orc8r_version)
+
+            if not agw_version_parsed:
+                logging.warning("Gateway version: %s not valid" % agw_version)
+
+            if not orc8r_version_parsed:
+                logging.warning("orchestrator version: %s not valid" % orc8r_version)
```

Comment:
Might be good to consider the case where the AGW is not connected to an Orc8r. In that case, what would the orc8r_version_parsed value be? and could we adjust the log in that case to be extra helpful?

Also,
```suggestion
                logging.warning("Orchestrator version: %s not valid" % orc8r_version)
```

Reply:
this value is read from gateway.mconfig file, so if the AGW is not connected to AGW, it will read whatever is there in AGW gateway.mconfig(depending on which gateway.mconfig file takes precedence)

Reply:
Yes I am wondering about the default value in the gateway.mconfig. (So the value in lte/gateway/configs/gateway.mconfig)  If the orc8r_version for the default mconfig is empty, then do we not log anything? (this seems fine to me). If the default version is some dummy value, then it might be confusing to for a user to see a log saying it's invalid.

Reply:
there is no default value defined in proto definition, and its not a optional field, so some value has to be defined for orc8r_version, looking at documentation its most likely gonna be a empty string(https://developers.google.com/protocol-buffers/docs/proto3#default)